### DISCUSSION TITLE
[Question][Hw4] Some questions regarding specification.

### DISCUSSION THREAD

**Discussion by Lee-Janggun** [prosyslab-classroom/cs424-program-reasoning#340] [Q&A]
I had a few questions regarding the details of HW4.

1. Does every example have `Bexp` and `Start`? If not, is there a good way to get all the possible
2. Are arguments, return value and the `Start` non terminals always Int type?
3. Will `Bexp` always be `bool`?
4. Does `Bexp` have any const or var terminals?
5. Are there are rules with depth greater than 2? That is, rules such as `(+ Start (+ Start Start))`?

**Comment by sujin0529** [comment]
Hi,
1. Every example have only `Start` and `Bexp` as non-terminal.
2. Yes. All arguments, return value and `Start` non-terminal are always Int type.
3. Yes. `Bexp` non-terminal is always Bool type.
4. If you ask if const and variable are given in `Bexp` grammar, it's not given.
5. No. Rules like `(+ Start (+ Start Start))` are not given in grammar. However, in program synthesis process it may require consideration.

Please ask me again if I misunderstood the question or if you have any more questions.

Thanks.

**Comment by Lee-Janggun** [comment]
Those were the answer I was looking for. Thank you for the help!
